plugin/black.vim

```diff
@@ -94,13 +94,44 @@ def _initialize_black_env(upgrade=False):
 if _initialize_black_env():
   import black
   import time
+  import toml
 
 def Black():
   start = time.time()
-  fast = bool(int(vim.eval("g:black_fast")))
-  line_length = int(vim.eval("g:black_linelength"))
+
+  path_pyproject_toml = black.abspath_pyproject_toml(vim.eval("fnamemodify(getcwd(), ':t')"))
+  try:
+    config_pyproject_toml = black.parse_pyproject_toml(path_pyproject_toml)
+  except Exception as exc:
+    config_pyproject_toml = {}
+
+  toml_line_length = config_pyproject_toml.get("line_length")
```

Comment:
It's not great that we have to hardcode all of these options here. Is there any way to avoid having separate code for each option? (Relatedly, #618 will change a few of these.)